modules/http/src/test/java/org/mule/module/http/internal/listener/HttpListenerTestCase.java

```diff
@@ -82,9 +97,87 @@ public RequestHandlerManager answer(InvocationOnMock invocation) throws Throwabl
     	assertThat(RequestContext.getEvent(), not(nullValue()));
     }
     
-    private void useInvalidPath(String listenerPath) throws InitialisationException
+    /**
+     * {@code host} header is not specified in HTTP 1.0
+     */
+    @Test
+    public void noHostHeaderOn10Request() throws Exception
     {
+        final AtomicReference<RequestHandler> requestHandlerRef = new AtomicReference<>();
+        when(mockHttpListenerConfig.addRequestHandler(any(ListenerRequestMatcher.class), any(RequestHandler.class))).then(new Answer<RequestHandlerManager>()
+        {
+            @Override
+            public RequestHandlerManager answer(InvocationOnMock invocation) throws Throwable
+            {
+                requestHandlerRef.set((RequestHandler) invocation.getArguments()[1]);
+                return null;
+            }
+        });
+        usePath("/");
+
+        HttpRequest request = buildGetRootRequest(HTTP_1_0);
+        HttpRequestContext requestContext = buildRequestContext(request);
+
+        HttpResponseReadyCallback responseCallback = mock(HttpResponseReadyCallback.class);
+        requestHandlerRef.get().handleRequest(requestContext, responseCallback);
+        assertResponse(responseCallback, 200);
+    }
 
+    @Test
+    public void noHostHeaderOn11Request() throws Exception
+    {
+        final AtomicReference<RequestHandler> requestHandlerRef = new AtomicReference<>();
+        when(mockHttpListenerConfig.addRequestHandler(any(ListenerRequestMatcher.class), any(RequestHandler.class))).then(new Answer<RequestHandlerManager>()
+        {
+            @Override
+            public RequestHandlerManager answer(InvocationOnMock invocation) throws Throwable
+            {
+                requestHandlerRef.set((RequestHandler) invocation.getArguments()[1]);
+                return null;
+            }
+        });
+        usePath("/");
+
+        HttpRequest request = buildGetRootRequest(HTTP_1_1);
+        HttpRequestContext requestContext = buildRequestContext(request);
+
+        HttpResponseReadyCallback responseCallback = mock(HttpResponseReadyCallback.class);
+        requestHandlerRef.get().handleRequest(requestContext, responseCallback);
+        assertResponse(responseCallback, 400);
```

Comment:
Can you verify the entire status line too? You should also be getting a particular body.